### DISCUSSION TITLE
Emitting OSC7 escape sequence on change_current_directory command execution?

### DISCUSSION THREAD

**Discussion by GregorLohaus** [helix-editor/helix#11310] [Ideas]
It would be great if the change_current_directory command emitted the OSC 7 escape sequence so terminal emulators that use it can take advantage of this. I've hacked this into my own build in the worst possible way so i wont create a pull request but it works so maybe someone can still use it as a reference for doing it the right way: https://github.com/GregorLohaus/helix/commit/b9957e1315c5fb709d667d982ee247e4bed2fa67

**Comment by the-mikedavis** [comment]
The implementation for the `:cd` command lives here: https://github.com/helix-editor/helix/blob/ef4a4ff3c558afff0917633876f11b586899934d/helix-term/src/commands/typed.rs#L1081-L1103

You could place the OSC7 parts after the call to `helix_stdx::env::set_current_working_dir(dir)` line. If you're looking for Helix to support OSC7 out of the box I would start with adding support for it in the terminal manipulation library we use: https://github.com/crossterm-rs/crossterm. Then we can upgrade our version of crossterm and use the support there. We try to not have bare escape sequences in the codebase - those belong in crossterm.
